polkadot/runtime/parachains/src/inclusion/mod.rs

```diff
@@ -979,6 +988,8 @@ impl<T: Config> Pallet<T> {
 	pub(crate) fn receive_upward_messages(para: ParaId, upward_messages: &[Vec<u8>]) -> Weight {
 		let bounded = upward_messages
 			.iter()
+			// Stop once we hit the `UMPSignal`` separator.
+			.take_while(|message| !message.is_empty())
```

Comment:
Would it be possible for someone with an account on the parachain to send an empty xcm message which would decode here as a separator. It could prevent the relay chain from executing all of the XCM messages.

Reply:
good question. we should try this

Reply:
I checked this, an XCM message is a `Vec` of XCM intructions so if it is empty it will still encode the 0 lenght so the resulting message is non-empty.